internal/pkg/userdata/userdata.go

```diff
@@ -252,12 +252,21 @@ func (data *UserData) WriteFiles() (err error) {
 }
 
 // Download initializes a UserData struct from a remote URL.
-func Download(url string) (data *UserData, err error) {
+func Download(url string, headers *map[string]string) (data *UserData, err error) {
```

Comment:
Introduces a headers input for the download function. Gcloud requires you pass a "Metadata-Flavor": "Google"" header

Reply:
Can we add a TODO here?

```
TODO(andrewrynhard): Implement functional options.